ci-metrics/README.md

```diff
@@ -0,0 +1,171 @@
+
+
+1. ####Run ansible to install ELK
+
+  >$ ansible-playbook install_elk.yaml
+
+
+1. ####Verify that ELK instance works
+
+    Access the port 80 on your localhost via your web browser and you should see the Kibana configuration page.
+
+  [http://localhost]
+
+
+## ELK Listener Jenkins job
+
+ELK Listener job listens on the CI message bus for messages matching this JMS selector:
+
+```
+(CI_TYPE = '{index}' AND method = 'build' AND scratch = FALSE) OR (CI_TYPE = 'ci-metricsdata')
+
+```
+
+These messages are stored directly into the ELK instance.
+
+The ELK Listener job requires:
+
+  - an existing ELK instance
+
+  - an existing PDC instance - see https://github.com/product-definition-center/
+
+  - a keytab for autentication to PDC via keberos
+
+### Installation of ELK Listener job
+
+1. ####Get CI Metrics files
+
+  >$ git clone https://github.com/RHQE/platform-ci.git
+
+  >$ cd platform-ci/ci-metrics
+
+1. ####Tweak [ci_listener.yaml](/ci-metrics/ci_listener.yaml)
+
+    Edit the file and:
+
+    - replace LINKTO_ELK_SERVER  with url to an ELK instance
+
+    - replace LINKTO_PDC url to PDC instance
+
+    Also make sure the environment variables KEYTAB and PRINCIPAL are defined in your Jenkins environment with details about your kerberos principal and keytab.
+
+1. ####Enable CI publisher for your Jenkins Job
+
+    To enable the job run jenkins-jobs from the Jenkins Job Builder on the file ci_listener.yaml.
+
+  >$ jenkins-jobs update ci_listener.yaml
+
+
+## Jenkins jobs with enabled CI publisher
+
+This section describes how to enable CI publisher for publishing CI metrics data from your jobs.
+
+The message types are defined in this document [https://url.corp.redhat.com/ci-metrics-message-types].
+
+The possible items of the message are defined in this document [https://url.corp.redhat.com/ci-metrics]
+
+### Howto enable CI publisher in your jobs in JJB yaml
+
+1. #### Make sure to inject the required variables that will be used by the CI publisher in the post build action
+
+    For example you can use the [EnvInject](https://wiki.jenkins-ci.org/display/JENKINS/EnvInject+Plugin) plugin to achive this.
+
+1. #### Enable CI publisher for your Jenkins Job
+
+    Add into your publisher section 'ci-publisher' definition like the one below. Make sure to correctly define the 'message-type' and all the relevant items in your message.
+
+```
+    publishers:
+      - ci-publisher:
+          message-type: 'Tier1TestingDone'
+          message-properties: |
+            CI_TYPE=ci-metricsdata
+          message-content: |
+            {{
+                "create_time": "$CREATE_TIME",
+                "tests": $TESTS,
+                "CI_tier": "$CI_TIER",
+                "owner": "$OWNER",
+                "build_type": "$BUILD",
+                "base_distro": "$BASE_DISTRO",
+                "completion_time": "$COMPLETION_TIME",
+                "component": "$name-$version-$release",
+                "job_link_back": "$JOB_URL",
```

Comment:
job_link_back has changed in the definition to be renamed to jenkins_job_url.  This is also reflected in PR #47 that I recently pushed